File: algorithms/400. Nth Digit/solution.py
class Solution(object):
    def findNthDigit(self, n):
        """
        :type n: int
        :rtype: int
        """
        # @StefanPochmann
        n -= 1
        for width in xrange(1, 11):
            first = 10 ** (width - 1)
            if n < 9 * width * first:
                return int(str(first + n / width)[n % width])
            n -= 9 * width * first

File: algorithms/400. Nth Digit/solution.cc
class Solution {
public:
    int findNthDigit(int n) {
        --n;
        int width = 1, first = 1;
        while ( n / 9 / width / first >= 1 ) {
            n -= 9 * width * first;
            ++width;
            first *= 10;
        }
        return to_string(first + n / width)[n % width] - '0';
    }
};
